alexander898911/main.js

```diff
@@ -0,0 +1,83 @@
+const STATUSES = {
+    TODO: 'To do',
+    'IN PROGRESS': 'In progress',
+    DONE: 'Done'
+};
+
+const PRIORITY={
+    LOW:'low',
+    HIGH:'high'
+};
+
+const todoList=[
+    {name: 'create a post', status: STATUSES["IN PROGRESS"], priority: PRIORITY.LOW}, 
+    {name: 'test', status: STATUSES.TODO, priority: PRIORITY.HIGH},
+];
+
+function addTask(taskName){
+    todoList.push({name: taskName, status: STATUSES.TODO, priority: PRIORITY.LOW});   
+}
+
+function changeStatus(taskName, taskStatus){
+    let foundObject = false;
+    for(let i = 0;i<todoList.length; i++){
+        if(todoList[i].name===taskName){
+            foundObject=true;
+            todoList[i].status = taskStatus;
+        }
+    }
+
+    if(foundObject===false){
+        console.log("Нет такой задачи");
+    }
+
+}
+function changePriority(taskName, taskPriority){
+    let foundObject = false;
+    for(let i = 0;i<todoList.length; i++){
+        if(todoList[i].name===taskName){
+            foundObject=true;
+            todoList[i].priority = taskPriority;
+        }
+    }
+    
+
+    if(foundObject===false){
+        console.log("Нет такой задачи");
+    }
+
+}
+function deleteTask(taskName){
+   
+        let task= todoList.indexOf(taskName);
+        todoList.splice(task,1);
+   
+}
+
+
+
+function ShowTask(){
+    let taskWithStatus = false;
+    const todoArray = todoList.filter(task => task.status===STATUSES[key]);
+    
+    todoArray.forEach(task=>{
+        console.log(`\t'${task.name}'` +'\t is ' +`${task.priority}`);
+        taskWithStatus = true;
+    });
+    if(!taskWithStatus){
+        console.log('\t-');
+    }
+   
+}
+function Show(){
+    for(key in STATUSES){
+        console.log(STATUSES[key] + ':');
+        ShowTask(STATUSES[key]);
```

Comment:
1 .Первую заглавную в JS принято использовать для имен конструкторов и классов.
2. В инициализации цикла рекомендуется использовать ключевые слова let(const), иначе она становится видна во всей программе(область видимости).
3. В 75 строке ты передаешь аргумент в showTask но в объявлении функции параметра нет, а функция отрабатывает потому что берет key из глобальной переменной объявленной (смотри пункт 2.)
```suggestion
function showTask(key){
    let taskWithStatus = false;
    const todoArray = todoList.filter(task => task.status === key);
    
    todoArray.forEach(task=>{
        console.log(`\t'${task.name}'` +'\t is ' +`${task.priority}`);
        taskWithStatus = true;
    });
    if(!taskWithStatus){
        console.log('\t-');
    }
   
}
function show(){
    for(const key in STATUSES){
        console.log(STATUSES[key] + ':');
        showTask(STATUSES[key]);
    }
}
```